crates/torii/server/src/proxy.rs

```diff
@@ -103,15 +105,19 @@ impl Proxy {
                         .collect::<Vec<HeaderName>>(),
                 );
 
-            let cors = match allowed_origins.as_slice() {
-                [origin] if origin == "*" => cors.allow_origin(AllowOrigin::mirror_request()),
-                origins => cors.allow_origin(
-                    origins.iter().map(|o| o.parse().expect("valid origin")).collect::<Vec<_>>(),
-                ),
-            };
+            let cors =
+                allowed_origins.clone().map(|allowed_origins| match allowed_origins.as_slice() {
+                    [origin] if origin == "*" => cors.allow_origin(AllowOrigin::mirror_request()),
+                    origins => cors.allow_origin(
+                        origins
+                            .iter()
+                            .map(|o| o.parse().expect("valid origin"))
```

Comment:
This parse will almost never fail, even with `--allowed-origins "fheihf egis g,9435"`. We may want to ensure they are valid URL if not equal to `*`. May be tackled later though depending on the priority.

Reply:
Oh yeah nice catch. We'll try to parse it as an URI first